src/statementsem.d

```diff
@@ -912,6 +912,16 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
                     fs.key = new VarDeclaration(loc, Type.tsize_t, idkey, null);
                     fs.key.storage_class |= STCtemp;
                 }
+                else
+                {
+                    if (!Type.tsize_t.implicitConvTo(fs.key.type))
+                    {
+                        fs.error("foreach: cannot implicitly convert the loop index from %s to %s",
```

Comment:
Indeed, I did use `error` because the implicit casts failing throw an error, but if you feel this should follow the deprecation path first I'll change it.